python/tvm/topi/nn/depthwise_conv2d.py

```diff
@@ -252,8 +253,18 @@ def depthwise_conv2d_nhwc(Input, Filter, stride, padding, dilation, out_dtype=No
         dilation_h, dilation_w = dilation
 
     batch, in_height, in_width, in_channel = Input.shape
+
+    dim = len(Input.shape) - 2
+
     # shape of dilated kernel
-    filter_height, filter_width, filter_channel, channel_multiplier = Filter.shape
+    if kernel_layout == "HWOI":
+        filter_height, filter_width, filter_channel, channel_multiplier = Filter.shape
+        kernel_permutation_to = [0, 1] + list(range(2, dim + 2))
+    elif kernel_layout == "HWIO":
+        filter_height, filter_width, channel_multiplier, filter_channel = Filter.shape
+        kernel_permutation_to = [dim + 1, dim] + list(range(dim))
+
+    kernel_permutation_from = np.argsort(kernel_permutation_to)
```

Comment:
Is there a benefit to defining in terms of `kernel_permutation_to` instead of directly defining `kernel_permutation_from`?

Reply:
I tried to follow [this](https://github.com/apache/tvm/blob/6782a35018d5df63486569f17296737d75eedc55/python/tvm/topi/nn/conv2d.py#L837) as much as possible so in the future we can add more features if needed. If you think this is not the best way I could change it.

Reply:
Ah, I see.  It looks like that implementation is trying to be more clever, and to identify the permutation by inspecting the string.  In this case, since we're only supporting two explicitly enabled shapes, I'd lean for defining a `kernel_permutation` directly for each one.

```python
if kernel_layout == 'HWOI':
    kernel_permutaiton = [0,1,2,3]
elif kernel_layout == 'HWIO':
    kernel_permutaiton = [0,1,3,2]
else:
    raise ValueError(f'Unsupported kernel layout: {kernel_layout}')
```

Reply:
That said, if there are many locations where the same kernel permutation definitions are used, it may be useful to pull it out into a common utility.

Reply:
Changed the way of permutation from your recommendation. @Lunderberg